bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NAWelcomeHomeHandler.java

```diff
@@ -97,96 +100,164 @@ protected NAWelcomeHome updateReadings() {
                     if (thing.isPresent()) {
                         NAWelcomeCameraHandler eventCamera = (NAWelcomeCameraHandler) thing.get();
-                        String streamUrl = eventCamera.getStreamURL(lastEvent.getVideoId());
+                        String streamUrl = eventCamera.getStreamURL(lastEvent.get().getVideoId());
                         if (streamUrl != null) {
                             return new StringType(streamUrl);
                         }
                     }
                 }
                 return UnDefType.UNDEF;
             case CHANNEL_WELCOME_EVENT_VIDEOSTATUS:
-                return lastEvent != null ? toStringType(lastEvent.getVideoStatus()) : UnDefType.UNDEF;
+                return lastEvent.map(e -> toStringType(e.getVideoStatus())).orElse(UnDefType.UNDEF);
             case CHANNEL_WELCOME_EVENT_ISARRIVAL:
-                return lastEvent != null ? lastEvent.getIsArrival() != null ? OnOffType.ON : OnOffType.OFF
-                        : UnDefType.UNDEF;
+                return lastEvent.map(e -> toOnOffType(e.getIsArrival())).orElse(UnDefType.UNDEF);
             case CHANNEL_WELCOME_EVENT_MESSAGE:
-                return lastEvent != null && lastEvent.getMessage() != null
-                        ? new StringType(lastEvent.getMessage().replace("<b>", "").replace("</b>", ""))
-                        : UnDefType.UNDEF;
+                return findEventMessage().map(
+                        m -> (State)new StringType(m.replace("<b>", "").replace("</b>", ""))
+                ).orElse(UnDefType.UNDEF);
             case CHANNEL_WELCOME_EVENT_SUBTYPE:
-                return lastEvent != null ? toDecimalType(lastEvent.getSubType()) : UnDefType.UNDEF;
+                return lastEvent.map(e -> toDecimalType(e.getSubType())).orElse(UnDefType.UNDEF);
         }
         return super.getNAThingProperty(channelId);
     }
 
+    @Override
+    protected void triggerChannelIfRequired(String channelId) {
+        if(isNewLastEvent) {
+            if(CHANNEL_CAMERA_EVENT.equals(channelId)) {
+                Set<String> detectedObjectTypes = findDetectedObjectTypes(lastEvent);
+                for(String detectedType: detectedObjectTypes) {
+                    triggerChannel(channelId, detectedType);
+                }
+            }
+        }
+        super.triggerChannelIfRequired(channelId);
+    }
+
+    private static Set<String> findDetectedObjectTypes(Optional<NAWelcomeEvent> eventOptional) {
+        Set<String> detectedObjectTypes = new TreeSet<>();
+        if(!eventOptional.isPresent()) {
+            return detectedObjectTypes;
+        }
+
+        NAWelcomeEvent event = eventOptional.get();
+
+        if(event.getPersonId() != null) {
+            detectedObjectTypes.add(CAMERA_EVENT_OPTION_HUMAN_DETECTED);
+        }
+
+        if(NAWebhookCameraEvent.EventTypeEnum.MOVEMENT.toString().equals(event.getType())) {
```

Comment:
Couldn't you use directly the EventTypeEnum to define the various events ?

Reply:
no, only MOVEMENT is available at EventTypeEnum. The others (human, animal and vehicle are at another enum). But I optimized the code a bit